### PR TITLE
feat: add domain type for client secret

### LINKED ISSUE #1357: [FEATURE] add domain type for client secret
### Feature Description

Create domain type for client secret

### Possible Implementation

Make type `ClietSecret(String)` and use it wherever applicable

### Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by tusharv01**
Add domain type for client secret in PaymentsSessionResponse

This commit introduces a new domain type, `ClientSecretDomain`, to handle the `client_secret` field in the `PaymentsSessionResponse` struct. The `ClientSecretDomain` type ensures proper encapsulation and security of sensitive information.

Changes made in this commit:
- Updated `PaymentsSessionResponse` struct to use `Secret<ClientSecretDomain, pii::ClientSecret>` for `client_secret` field.
- Implemented serialization and deserialization for `ClientSecretDomain`.
- Updated the `From` implementation to convert `PaymentsSessionRequest` into `PaymentsSessionResponse` with the appropriate `ClientSecretDomain`.

This enhancement improves the security and maintainability of the codebase by properly handling client secrets in a more structured and encapsulated manner. It also aligns with the PR #1357, addressing the feature request to create a domain type for the client secret.

Test cases have been added to ensure the proper serialization and deserialization of the `ClientSecretDomain`.

Closes #1357

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -13,6 +13,62 @@ use utoipa::ToSchema;
...
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        Ok(ClientSecretDomain::new(s))
+    }
+}
+
+impl Serialize for ClientSecretDomain {
```
Comment (Abhicodes-crypto) [inline crates/api_models/src/payments.rs:45]: In serialise you can concat payment_id and secret and give out a string.

File: crates/api_models/src/payments.rs [hunk 2]
```diff
@@ -13,6 +13,62 @@ use utoipa::ToSchema;
...
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        self.domain.serialize(serializer)
+    }
+}
+
+#[derive(Default, Debug, serde::Serialize, Clone, ToSchema)]
+pub struct PaymentsSessionResponse {
```
Comment (Abhicodes-crypto) [inline crates/api_models/src/payments.rs:55]: Why was this added ?

**Review by Abhicodes-crypto — CHANGES_REQUESTED** [review]

**Comment by VedantKhairnar** [comment]
Closing due to inactivity. Feel free to open it if you are interested and assigned the corresponding issue.
Thanks.

**Comment by tusharv01** [comment]
Hello can we open this PR please as i am interested to work on it.

**Comment by tusharv01** [comment]
@VedantKhairnar @SanchithHegde

**Comment by SanchithHegde** [comment]
@tusharv01 I believe #1357 was closed in #3557.

**Pull request closed without merging**